### ISSUE TITLE
Could ref be used to optimize multiple field access?

### ISSUE THREAD

**Issue by wmjordan** [u2uconsult/codeanalyzers#8]
Inspired by **U2U1015**, I wondered whether the following code could be optimized.

````C#
if (Interlocked.Increment(ref StaticCode.Value) > 0) {
    // do something
    Interlocked.Decrement(ref StaticCode.Value);
}

static class StaticCode {
    internal static int Value;
}
````

It is similar to _U2U1015_. Could the following code run faster than the previous one by avoiding addressing the static type twice?

````C#
ref int i = ref StaticCode.Value;
if (Interlocked.Increment(ref i) > 0) {
    // do something
    Interlocked.Decrement(ref i);
}
````

**Comment by KrisVandermotten** [comment]
It might, but probably not.

First of all, getting the address of a static field is simpler than getting the address of an array element. The cost is low anyway, in fact, after (jit) compilation, it might actually be zero. In this case, compared to the cost of the `Interlocked` operations, it is definitely negligible.

Secondly, **U2U1015** is reported only in loops, because of the multiplication effect. Without a loop, it's probably not worth doing this.

Finally, it is very difficult to *guess* performance impact of small changes like this.

You might want to read Eric Lippert's excellent [performance rant](https://ericlippert.com/2012/12/17/performance-rant/).

**Comment by wmjordan** [comment]
Thank you for the clarification. I fully agree with you.
